src/lists.c

```diff
@@ -146,7 +146,6 @@ static Obj AttrLENGTH(Obj self, Obj list)
     /* internal list types                                                 */
 #ifdef HPCGAP
     ReadGuard(list);
```

Comment:
Why is this `ReadGuard` needed, though?

Reply:
Yes, that can be removed, too, now I look at it. It used to be paired with an `ImpliedWriteGuard()` in order to inform data flow analysis in Ward. On its own, it doesn't serve a purpose anymore (and given how the code has changed, probably hasn't for a while).